test/e2e/reloadinterval_test.go

```diff
@@ -0,0 +1,94 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestReloadInterval(t *testing.T) {
+	t.Parallel()
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "reloadinterval-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	t.Logf("waiting for ingresscontroller %v", icName)
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, icName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller router deployment: %v", err)
+	}
+
+	for _, testCase := range []struct {
+		description    string
+		reloadInterval metav1.Duration
+		expectedEnvVar string
+	}{
+		// set Spec.TuningOptions.ReloadInterval to a nondefault value within allowed bounds
+		{"set reloadInterval 100s", metav1.Duration{100 * time.Second}, "100s"},
+		{"set reloadInterval 1m", metav1.Duration{1 * time.Minute}, "1m"},
+
+		// set ReloadInterval to 0, which should give default of 5s
+		{"set reloadInterval default", metav1.Duration{0 * time.Second}, "5s"},
+
+		// set ReloadInterval to a value outside of allowed bounds
+		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "120s"},
+		{"set reloadInterval 0.5s", metav1.Duration{500 * time.Millisecond}, "1s"},
```

Comment:
0.5ms is rejected by the API validation.  "130s" gets capped to 120 seconds and then converted to "2m".  
```suggestion
		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "2m"},
```